### PR TITLE
Reland the decision task

### PR DESCRIPTION
**Pull request by jgraham**

### CODE REVIEW

**Comment by Hexcles** [comment]
> There are three known bugs:
> * Commits landing in short consecutive on master would cause job failures. Potential fix #20406
> * New PR jobs are failing because InterpreterError at template.tasks[0].payload.command[3]: object has no property after
> * Stability/unittest jobs are missing.

**Comment by Hexcles** [comment]
Proposal: maybe we can check in the full expected JSON payloads for both PR and master, and verify against those baselines instead of only validating against the schema.

Chrome infra does this very often.

**Comment by jgraham** [comment]
I think that's probably the right thing to do. I'm surprised by the task submission failure; I don't see why that would happen now but not in the previous PR or in all PRs.

**Comment by Hexcles** [comment]
I'm wondering if this is yet another case of partial dependency on the branch base.

I suspect if you rebase this branch now, we won't see the submission error any more.

**Comment by jgraham** [comment]
Oh, I guess it succeeds for subsequent pushes?

**Comment by jgraham** [comment]
@Hexcles I think this is ready for another round of review

File: tools/ci/tc/tests/test_valid.py [hunk 1]
```diff
@@ -71,3 +73,74 @@ def test_verify_payload():
             except Exception as e:
                 print("Validation failed for task '%s':\n%s" % (name, json.dumps(task_data, indent=2)))
                 raise e
+
+
+@pytest.mark.xfail(sys.version_info.major == 2,
+                   reason="taskcluster library has an encoding bug "
+                   "https://github.com/taskcluster/json-e/issues/338")
```
Comment (gsnedders) [inline tools/ci/tc/tests/test_valid.py]: We should expect this to start failing on master at some point, given we don't pin json-e, right? 🙁

Comment (jgraham) [reply tools/ci/tc/tests/test_valid.py]: I agree that we should pin all the requirements in a requirements.txt fle, but I don't think that should block this.

**Review by gsnedders — APPROVED** [review]
but should probably wait for @Hexcles too

File: tools/ci/tc/tasks/test.yml [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+            channel: nightly
+        - vars:
+            channel: beta
+        - vars:
+            channel: stable
+      do:
+        download-firefox-${vars.channel}:
+          use:
+            - wpt-base
+          command: "./wpt install --download-only --destination /home/test/artifacts/ --channel=${vars.channel} firefox browser"
```
Comment (Hexcles) [inline tools/ci/tc/tasks/test.yml]: IIUC, the intention is to download the browser binary in one task and reuse it, but I don't quite understand how the plumbing works. Is `/home/test/artifacts` automatically shared across tasks? Where do we instruct the tasks to find binaries from artifacts?

File: tools/wpt/browser.py [hunk 3]
```diff
@@ -47,6 +47,11 @@ class Browser(object):
     def __init__(self, logger):
         self.logger = logger
 
+    @abstractmethod
```
Comment (Hexcles) [inline tools/wpt/browser.py]: I would really wish splitting this out into a separate PR, but I won't block you on this.

File: tools/ci/tc/tasks/test.yml (hunk 2 above)
Comment (jgraham) [reply tools/ci/tc/tasks/test.yml]: It doesn't yet :) The idea was to prove out the task dependency part in this PR and actually use the binary as a followup (it probably means having some way to specify what the URL is and something to get that URL).

**Review by Hexcles — APPROVED** [review]
Alright in that case, this LGTM. I don't have very high confidence, but higher than last time because of the tests :)

I'm not fully available this week so I'm volunteering @stephenmcgruer for monitoring :)

**Comment by jgraham** [comment]
I wouldn't say I have high confidence either. But the amount that can go wrong is relatiely low (in the sense it's fairly easy to backout).

Do you have things that I could do to improve your confidence?

**Comment by Hexcles** [comment]
If we could make this PR smaller, perhaps? Like splitting out the `download` stuff.

**Comment by jgraham** [comment]
https://github.com/web-platform-tests/wpt/pull/20463 for the download stuff

**Pull request merged**